app/src/main/java/ie/teamchile/smartapp/util/EnvironmentChecker.java

```diff
@@ -0,0 +1,115 @@
+package ie.teamchile.smartapp.util;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.Signature;
+import android.os.Build;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Created by user on 9/12/15.
+ */
+public class EnvironmentChecker {
+    private static String cert = "";
+
+    public static boolean isCertInvalid(Context context) {
+        String fingerprintMD5 = "";
+        Signature[] sigs = new Signature[0];
+        try {
+            sigs = context.getPackageManager().getPackageInfo(context.getPackageName(),
+                    PackageManager.GET_SIGNATURES).signatures;
+
+            for (Signature sig : sigs) {
+                byte[] hexBytes = sig.toByteArray();
+                MessageDigest digest = MessageDigest.getInstance("MD5");
+
+                byte[] md5digest = new byte[0];
+
+                if (digest != null) {
+                    md5digest = digest.digest(hexBytes);
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < md5digest.length; ++i) {
+                        sb.append((Integer.toHexString((md5digest[i] & 0xFF) | 0x100)).substring(1, 3));
+                    }
+                    fingerprintMD5 = sb.toString();
+                }
+            }
+
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+        return !fingerprintMD5.equals(cert);
+    }
+
+    public static boolean isDeviceRooted() {
+        return (checkRootMethod1() || checkRootMethod2() || checkRootMethod3());
+    }
+
+    private static boolean checkRootMethod1() {
+        String buildTags = android.os.Build.TAGS;
+        return buildTags != null && buildTags.contains("test-keys");
```

Comment:
The 'test-key' here relate to the keys that are used to build the ROM. Stock Google ROMs are built with 'release-keys'. So it the build tags contain a key called 'test-keys' it could mean that the ROM is a custom built ROM.